### PR TITLE
[Hub Generated] Review request for Microsoft.DocumentDB to add version preview/2022-02-15-preview

### PR DESCRIPTION
**Pull request by amar-sagare**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/amar-sagare/azure-rest-api-specs/amarsa-cosmos-db-Microsoft.DocumentDB-2022-02-15-preview...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month : **March 2022**  
  4. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. **3/3/2022**
  5. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by wmengmsft — APPROVED** [review]

**Comment by mentat9** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/mongodbDatabases/{databaseName}/collections/{containerName}/partitionMerge": {

collectionName?

---
Refers to: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json:3518 in a6b0e0c. [](commit_id = a6b0e0caaba7f843ae63cc92eb71113d0471315e, deletion_comment = False)

**Comment by mentat9** [comment]
>         "isDryRun": {

ARM recommends string enums over boolean. Generally enums make better properties than booleans. They are more descriptive, flexible and future-proof while being easier for customers to discover, understand and use. Ref: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review).

Consider using an enum, maybe something like executionType { "Report", "Update" }. This type of property is easier for customers to understand and use.

---
Refers to: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json:9906 in a6b0e0c. [](commit_id = a6b0e0caaba7f843ae63cc92eb71113d0471315e, deletion_comment = False)

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/notebook.json
Comment (mentat9) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/notebook.json:71]: >../../preview/2022-02-15-preview/cosmos-db.json#/definitions/ErrorResponse [](http://example.com/codeflow?start=23&length=74)

Consider moving to common-types for these common types.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/notebook.json
Comment (mentat9) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/notebook.json:368]: >../../preview/2022-02-15-preview/cosmos-db.json#/definitions/ARMProxyResource [](http://example.com/codeflow?start=19&length=77)

Consider using Resource from common-types v2 or v3 types.json (depending on whether you want/need systemData).

**Comment by amar-sagare** [comment]
>         "isDryRun": {

The property represents a boolean value whether it is a simulation or it isn't. There isn't a third possibility here.

---
In reply to: [1062541292](https://github.com/Azure/azure-rest-api-specs/pull/18087#issuecomment-1062541292) [](http://example.com/codeflow?ancestors=1062541292,1062541292)

---
Refers to: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json:9906 in a6b0e0c. [](commit_id = a6b0e0caaba7f843ae63cc92eb71113d0471315e, deletion_comment = False)

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/notebook.json
Comment (amar-sagare) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/notebook.json:368]: These schemas are the same as the previous ones, there is no change here. Can we take this as an action item so that it gets fixed in the next PR?

**Comment by mentat9** [comment]
> > ```
> >     "isDryRun": {
> > ```
> 
> ARM recommends string enums over boolean. Generally enums make better properties than booleans. They are more descriptive, flexible and future-proof while being easier for customers to discover, understand and use. Ref: https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review.
> 
> Consider using an enum, maybe something like executionType { "Report", "Update" }. This type of property is easier for customers to understand and use.
> 
> In reply to: [1062541292](https://github.com/Azure/azure-rest-api-specs/pull/18087#issuecomment-1062541292)
> 
> In reply to: [1062541292](https://github.com/Azure/azure-rest-api-specs/pull/18087#issuecomment-1062541292)
> 
> In reply to: [1062541292](https://github.com/Azure/azure-rest-api-specs/pull/18087#issuecomment-1062541292)
> 
> Refers to: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json:9906 in [a6b0e0c](https://github.com/Azure/azure-rest-api-specs/commit/a6b0e0caaba7f843ae63cc92eb71113d0471315e). [](commit_id = [a6b0e0c](https://github.com/Azure/azure-rest-api-specs/commit/a6b0e0caaba7f843ae63cc92eb71113d0471315e), deletion_comment = False)

I won't block approval on this, but still recommend enums even for properties with only two values.

**Comment by mentat9** [comment]
Signed off with comments from ARM side.

**Comment by amar-sagare** [comment]
@changlong-liu , Can you please verify that the swagger SDK gate failure is benign?

**Comment by changlong-liu** [comment]
Adding @qiaozha  and @ArcturusZhang for JS/GO SDK breaking change review.

**Comment by changlong-liu** [comment]
Hi @amar-sagare ,  since "API Readiness Check " failed, the new api version has not been deployed, right?

Generally, swagger can only get merged after the service is ready.

**Comment by amar-sagare** [comment]
new api version is not yet deployed. I wanted a review for the SDK breaking changes so that we know the swagger is fine.


---
In reply to: [1064686302](https://github.com/Azure/azure-rest-api-specs/pull/18087#issuecomment-1064686302) [](http://example.com/codeflow?ancestors=1064686302,1064686302)

**Comment by vinhms** [comment]
hi @changlong-liu , I've updated the PR with one more commit.  Can you please review it one more time?

Also, the validations didn't seem to be working and they have been in the "Expected - waiting for status to be reported" for almost a day now.  Do you know how to fix it?

**Comment by vinhms** [comment]
hi @changlong-liu , please review the PR again and reapprove.

I've updated the PR to fix the prettycheck and cross-version issue.  The two remaining failed checks are same as before my change and was approved.

@amar-sagare  FYI

**Comment by changlong-liu** [comment]
Hi @vinhms , the "API Readiness Check" is failed. So the service has not been deployed, has it?
Generally the service should be deployed before swagger merging

**Comment by vinhms** [comment]
> Hi @vinhms , the "API Readiness Check" is failed. So the service has not been deployed, has it? Generally the service should be deployed before swagger merging

hi @changlong-liu this version will be deployed soon (within the week).  We won't merge until then.  We want to have it reviewed so we know the swagger change is fine.

**Comment by changlong-liu** [comment]
> 
> hi @changlong-liu this version will be deployed soon (within the week). We won't merge until then. We want to have it reviewed so we know the swagger change is fine.

Hi @vinhms , this looks good to me, added Approved label

File: specification/cosmos-db/resource-manager/readme.md
Comment (amar-sagare) [inline specification/cosmos-db/resource-manager/readme.md:30]: @changlong-liu , can you please merge the PR.

**Pull request merged**
